code/test/tokens.js

```diff
@@ -0,0 +1,472 @@
+const TestUtil = require('./util')
+const CheckUtil = require('./checkUtil')
+const unit = require('ethjs-unit')
+const ONEUtil = require('../lib/util')
+const ONEDebugger = require('../lib/debug')
+const ONE = require('../lib/onewallet')
+const ONEConstants = require('../lib/constants')
+const Flow = require('../lib/api/flow')
+const ONEWallet = require('../lib/onewallet')
+const crypto = require('crypto')
+const BN = require('bn.js')
+const { getIdentificationKey } = require('../lib/util')
+
+const INTERVAL = 30000
+const DURATION = INTERVAL * 12
+const SLOT_SIZE = 1
+const EFFECTIVE_TIME = Math.floor(Date.now() / INTERVAL / 6) * INTERVAL * 6 - DURATION / 2
+const ONE_CENT = unit.toWei('0.01', 'ether')
+// const HALF_DIME = unit.toWei('0.05', 'ether')
+// const ONE_DIME = unit.toWei('0.1', 'ether')
+// const ONE_ETH = unit.toWei('1', 'ether')
+const TEN_ETH = unit.toWei('10', 'ether')
+
+const Logger = TestUtil.Logger
+const Debugger = ONEDebugger(Logger)
+
+contract('ONEWallet', (accounts) => {
+  let snapshotId
+  beforeEach(async function () {
+    snapshotId = await TestUtil.snapshot()
+    await TestUtil.init()
+  })
+
+  afterEach(async function () {
+    await TestUtil.revert(snapshotId)
+  })
+
+  // Transfer Native Token to external wallet
+  it('Wallet_CommitReveal: Native Token Transfer must commit and reveal successfully', async () => {
+    // Create Wallets and tokens
+    const alice = await makeWallet(1, accounts[1])
```

Comment:
I suggest using a unique number specific to the test (e.g. hash of the name, then converted to BN) for salt. The salt will determine the wallet address, so a unique value ensures it doesn't collide with any existing wallet

Reply:
Agreed - will implement

Reply:
This has now been modified to create new accounts using a unique identifier
TT - Two Character Test Identifier (one per test file)
XX - Sub Test one per test 
99 - Wallet Identifier with test
Account creation logic
`   let alice = await CheckUtil.makeWallet('TT-ERC20-1', accounts[0], EFFECTIVE_TIME)`
Wallet Creation Function - note `effectiveTime` (the wallets effective creation time) is defined in the test and passed.
```
// makeWallet uses an index and unlocked web3.eth.account and creates and funds a ONEwallet
const makeWallet = async (salt, deployer, effectiveTime) => {
  const lastResortAccount = web3.eth.accounts.create()
  const { wallet, seed, hseed, root, client: { layers } } = await TestUtil.createWallet({
    salt: new BN(ONEUtil.keccak(salt)),
    effectiveTime: effectiveTime,
    duration: DURATION,
    maxOperationsPerInterval: SLOT_SIZE,
    lastResortAddress: lastResortAccount.address,
    spendingLimit: ONE_ETH
  })
  // Fund wallet
  await web3.eth.sendTransaction({
    from: deployer,
    to: wallet.address,
    value: HALF_ETH
  })
  const currentState = await getONEWalletState(wallet)
  return { wallet, seed, hseed, root, layers, lastResortAddress: lastResortAccount.address, currentState }
}
```